app/models/user.rb

```diff
@@ -66,7 +68,7 @@ def unset_primary_email
   def set_next_email_as_primary
     return false unless default_email_record
 
-    default_email_record.update_attributes(primary: true)
+    default_email_record.destroyed? || default_email_record.update_attributes(primary: true)
```

Comment:
maybe we should change the behaviour of default_email_record never to return a destroyed record?